amgut/lib/data_access/ag_data_access.py

```diff
@@ -969,3 +969,303 @@ def is_deposited_ebi(self, barcode):
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No emails found.')
+            return info[0][0]
+
+    def get_random_barcode(self, deposited=True):
+        """ Returns randomly chosen barcode.
+        For unit testing only!
+
+        Parameters
+        ----------
+        deposited : boolean
+            If true, pick a deposited barcode. Default = True
+
+        Returns
+        -------
+        str: barcode
+            Example: '000032951'
+
+        Raises
+        ------
+        ValueError
+            If no barcodes can be found in the DB."""
+        depos = 'TRUE'
+        if deposited is False:
+            depos = 'FALSE'
+        with TRN:
+            sql = """SELECT barcode
+                     FROM ag.ag_kit_barcodes
+                     WHERE deposited=%s
+                     LIMIT 1"""
+            TRN.add(sql, [depos])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No barcodes found.')
+            return info[0][0]
+
+    def get_email_from_ag_login_id(self, ag_login_id):
+        """ Returns email for a given ag_login_id.
+        For unit testing only!
+
+        Parameters
+        ----------
+        ag_login_id : str
+            Existing ag_login_id.
+
+        Returns
+        -------
+        str: email
+            Example: 'xX/tEv7O+T@6Ri7C.)LO'
+
+        Raises
+        ------
+        ValueError
+            If ag_login_id is not in DB.
+        """
+        with TRN:
+            sql = """SELECT email
+                     FROM ag.ag_login
+                     WHERE ag_login_id=%s"""
+            TRN.add(sql, [ag_login_id])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No emails found.')
+            return info[0][0]
+
+    def get_supplied_kit_id(self, ag_login_id):
+        """ Returns supplied_kit_id for a given ag_login_id.
+        For unit testing only!
+
+        Parameters
+        ----------
+        ag_login_id : str
+            Existing ag_login_id.
+
+        Returns
+        -------
+        str
+            The supplied_kit_id for the given ag_login_id.
+            Example: 'DokBF'
+
+        Raises
+        ------
+        ValueError
+            If ag_login_id is not in DB.
+        """
+        with TRN:
+            sql = """SELECT supplied_kit_id
+                     FROM ag.ag_kit
+                     WHERE ag_login_id = %s"""
+            TRN.add(sql, [ag_login_id])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('ag_login_id not in database: %s' %
+                                 ag_login_id)
+            return info[0][0]
+
+    def get_participant_names_from_ag_login_id(self, ag_login_id):
+        """ Returns all participant_name(s) for a given ag_login_id.
+        For unit testing only!
```

Comment:
I checked with existing methods, there should be no overlap. The cleanest solution would be to move those functions into their own file but this is out of scope for this PR.